app/addons/documents/index-editor/actions.js

```diff
@@ -15,142 +15,282 @@ define([
+      if (viewInfo.originalDesignDocName !== viewInfo.designDocId) {
+        var oldDesignDoc = findDesignDoc(viewInfo.designDocs, viewInfo.originalDesignDocName);
+        safeDeleteView(oldDesignDoc, viewInfo.designDocs, viewInfo.database, viewInfo.originalViewName);
+      }
 
-        if (_.any([viewInfo.designDocChanged, viewInfo.hasViewNameChanged, viewInfo.newDesignDoc, viewInfo.newView])) {
-          FauxtonAPI.dispatch({ type: ActionTypes.VIEW_SAVED });
-          var fragment = FauxtonAPI.urls('view', 'showNewlySavedView', viewInfo.database.safeID(), designDoc.safeID(), app.utils.safeURLName(viewInfo.viewName));
-          FauxtonAPI.navigate(fragment, { trigger: true });
-        } else {
-          this.updateDesignDoc(designDoc);
-        }
-
-        // this can be removed after the Views are on their own page
-        IndexResultsActions.reloadResultsList();
-      }.bind(this));
-    },
+      if (viewInfo.designDocId === 'new-doc') {
+        addDesignDoc(designDoc);
+      }
 
-    updateDesignDoc: function (designDoc) {
-      FauxtonAPI.dispatch({
-        type: ActionTypes.VIEW_UPDATE_DESIGN_DOC,
-        designDoc: designDoc.toJSON()
-      });
-    },
+      FauxtonAPI.dispatch({ type: ActionTypes.VIEW_SAVED });
+      var fragment = FauxtonAPI.urls('view', 'showView', viewInfo.database.safeID(), designDoc.safeID(), app.utils.safeURLName(viewInfo.viewName));
+      FauxtonAPI.navigate(fragment, { trigger: true });
+    });
+  }
 
-    deleteView: function (options) {
-      var viewName = options.viewName;
-      var database = options.database;
-      var designDoc = ActionHelpers.findDesignDoc(options.designDocs, options.designDocId);
-      var promise;
+  function addDesignDoc (designDoc) {
+    FauxtonAPI.dispatch({
+      type: ActionTypes.VIEW_ADD_DESIGN_DOC,
+      designDoc: designDoc.toJSON()
+    });
+  }
 
-      designDoc.removeDdocView(viewName);
+  function deleteView (options) {
 
-      if (designDoc.hasViews()) {
-        promise = designDoc.save();
-      } else {
-        promise = designDoc.destroy();
-      }
+    var onSuccess = function () {
 
-      promise.then(function () {
-        var url = FauxtonAPI.urls('allDocs', 'app', database.safeID(), '?limit=' + FauxtonAPI.constants.DATABASES.DOCUMENT_LIMIT);
+      // if the user was on the index that was just deleted, redirect them back to all docs
+      if (options.isOnIndex) {
+        var url = FauxtonAPI.urls('allDocs', 'app', options.database.safeID(), '?limit=' + FauxtonAPI.constants.DATABASES.DOCUMENT_LIMIT);
         FauxtonAPI.navigate(url);
-        FauxtonAPI.triggerRouteEvent('reloadDesignDocs');
-      });
-    },
+      }
+
+      FauxtonAPI.triggerRouteEvent('reloadDesignDocs');
 
-    updateMapCode: function (code) {
-      FauxtonAPI.dispatch({
-        type: ActionTypes.VIEW_UPDATE_MAP_CODE,
-        code: code
+      FauxtonAPI.addNotification({
+        msg: 'The <code>' + options.indexName + '</code> view has been deleted.',
```

Comment:
we need to escape `options.indexName` to make sure to open no XSS vector

something like
```js
'The <code>' + _.escape(options.indexName) + '</code> view has been deleted.',
```